security_monkey/auditors/s3.py

```diff
@@ -80,36 +80,28 @@ def check_policy(self, s3_item):
             self.inspect_policy_conditionals(statement, s3_item)
 
     def inspect_policy_allow_all(self, statement, s3_item):
-        if statement['Effect'] == "Allow":
-            if statement['Principal'] == "*":
+        if statement.get('Effect') == "Allow":
+            principal = statement.get('Principal')
+            if isinstance(principal, basestring) and principal == "*":
                 message = "POLICY - This Policy Allows Access From Anyone."
                 self.add_issue(10, message, s3_item)
                 return
 
-            if 'AWS' in statement['Principal']:
-                if statement['Principal']['AWS'] == "*":
-                    message = "POLICY - This Policy Allows Access From Anyone."
-                    self.add_issue(10, message, s3_item)
-                    return
+            if isinstance(principal, dict) and principal.get('AWS') == "*":
+                message = "POLICY - This Policy Allows Access From Anyone."
+                self.add_issue(10, message, s3_item)
+                return
 
     def inspect_policy_cross_account(self, statement, s3_item, complained):
         try:
-            if 'Effect' in statement:
-                effect = statement['Effect']
-                if effect == 'Allow':
-                    if 'Principal' in statement:
-                        principal = statement["Principal"]
-                        if type(principal) is dict and 'AWS' in principal:
-                            aws_entries = principal["AWS"]
-                            if type(aws_entries) is str or type(aws_entries) is unicode:
-                                if aws_entries[0:26] not in complained:
-                                    self.processCrossAccount(aws_entries, s3_item)
-                                    complained.append(aws_entries[0:26])
-                            else:
-                                for aws_entry in aws_entries:
-                                    if aws_entry[0:26] not in complained:
-                                        self.processCrossAccount(aws_entry, s3_item)
-                                        complained.append(aws_entry[0:26])
+            if statement.get('Effect') == 'Allow' and isinstance(statement.get("Principal"), dict):
+                aws_entries = statement["Principal"].get("AWS", [])
+                if isinstance(aws_entries, basestring):
+                    aws_entries = [aws_entries]
+                for aws_entry in aws_entries:
+                    if aws_entry[0:26] not in complained:
```

Comment:
Sounds good.

I did make some changes to the auditor that got merged in and is now causing conflicts.  Can you rebase your PR off of the latest develop and resolve the conflicts?